pkg/interactive/validation.go

```diff
@@ -187,20 +188,25 @@ func AvailabilityZonesCountValidator(multiAZ bool) Validator {
 	}
 }
 
-func MachinePoolRootDiskSizeValidator(val interface{}) error {
-	// We expect GigiByte as the unit for the root volume size
+func MachinePoolRootDiskSizeValidator(version string) Validator {
+	return func(val interface{}) error {
+		// We expect GigiByte as the unit for the root volume size
 
-	// Validate the worker root volume size is an integer
-	machinePoolRootDiskSize, ok := val.(string)
-	if !ok {
-		return fmt.Errorf("machine pool root disk size must be an string, got %T", machinePoolRootDiskSize)
-	}
+		// Validate the worker root volume size is an integer
+		machinePoolRootDiskSize, ok := val.(string)
+		if !ok {
+			return fmt.Errorf("machine pool root disk size must be an string, got %T", machinePoolRootDiskSize)
+		}
 
-	// parse it to validate it is a valid unit
-	_, err := ocm.ParseDiskSizeToGigibyte(machinePoolRootDiskSize)
-	if err != nil {
-		return fmt.Errorf("failed to parse machine pool root disk size: %v", err)
+		// parse it to validate it is a valid unit
+		size, err := ocm.ParseDiskSizeToGigibyte(machinePoolRootDiskSize)
+		if err != nil {
+			return fmt.Errorf("failed to parse machine pool root disk size: %v", err)
+		}
+		err = diskValidator.ValidateMachinePoolRootDiskSize(version, size)
```

Comment:
NP
```suggestion
		return diskValidator.ValidateMachinePoolRootDiskSize(version, size)
```

Reply:
@oriAdler 
thanks for the review. Made the refactoring, can you help to review it again?